packages/model-viewer/src/features/ar.ts

```diff
@@ -340,9 +349,50 @@ configuration or device capabilities');
       const img = document.createElement('img');
       anchor.appendChild(img);
       anchor.setAttribute('href', modelUrl.toString());
+      if (this.arAutogenerateUsdz) {
+        anchor.setAttribute('download', 'model.usdz');
+      }
       anchor.click();
       anchor.removeChild(img);
     }
+
+    [$onModelLoad]() {
+      super[$onModelLoad]();
+      if (this.arAutogenerateUsdz && !!this.iosSrc) {
+        URL.revokeObjectURL(this.iosSrc);
+        this.iosSrc = null;
+      }
+    }
+
+    async prepareUSDZ() {
+      const updateSourceProgress = this[$progressTracker].beginActivity();
+      const scene = this[$scene];
+
+      const shadow = scene.shadow;
+      let visible = false;
+
+      // Remove shadow from export
+      if (shadow != null) {
+        visible = shadow.visible;
+        shadow.visible = false;
+      }
+
+      updateSourceProgress(0.2);
+
+      const exporter = new USDZExporter();
+      const arraybuffer = await exporter.parse(scene.modelContainer);
+      const blob = new Blob([arraybuffer], {
+        type: 'application/octet-stream',
+      });
+
+      this.iosSrc = URL.createObjectURL(blob);
```

Comment:
I'm not sure I like this internal detail being publicly visible; maybe better to create a new private symbol instead.

Reply:
You mean showing the loading bar? I was initially thinking about showing some custom poster but then decided to just show some simulated (0.2) progress. On latest iPhone it is quite immediate to export a simple scene, but on older devices or when the scene is quite complex, it might require more time.

Reply:
No, the loading bar is great. It's just that `this.iosSrc` is part of our public API. If someone replaced it without revoking it, bad things would happen. I just mean we should store our blob as a private member instead.

Reply:
Or the Blob is garbage collected after URL.revokeObjectURL is called?

Reply:
I think it is probably still a good thing to store blob url in separate member. Let me make those changes.